src/Apps/Sale/Components/SaleArtworks.tsx

```diff
@@ -13,11 +13,13 @@ import { SaleArtworksFilter_viewer$data } from "__generated__/SaleArtworksFilter
 interface SaleArtworkFilterProps {
   relay: RelayRefetchProp
   viewer: SaleArtworksFilter_viewer$data
+  featuredKeywords?: readonly string[]
```

Comment:
If we're ever defining a field in a relay query that we're then using in the UI (meaning, in some fashion its being passed through as props) then we try to avoid mirroring / duplicating the interface types, since there's already a typescript file that's been generated by relay, by virtue of us writing the query. This is all very confusing, but can explain via an example:

```tsx
// Avoid 
interface AppProps {
  sale: {
    name: string
  }
}

// Good

import { App_sale$data } from '__generated__/App_sale.graphql'

interface AppProps {
  sale: App_sale$data
}

const App: React.FC<AppProps> = ({ sale }) => {

}

export const AppFragmentContainer = createFragmentContainer(App, {
  sale: graphql`
    fragment App_sale on Sale {
      name 
    }
  `
})

So here, to avoid the strange `readonly` stuff, can just use the type that's generated and passed in directly:

```tsx
interface SaleArtworkFilterProps {
  ...
  featuredKeywords: SaleApp_sale$data['featuredKeywords']
}
```

Doing this also helps avoid having to do the `| undefined | null` dance.

All that said, in the end (since this is consumed way down in the ArtworkFilter, beyond a product area surface boundary, and is not a straightforward relay component) unless you want to provide a union type there it might be more straight forward to do things as you are. Just know that typically we try to avoid this and always lean on the generated typescript files.

Reply:
Exactly the kind of advice I need 🙏